### ISSUE TITLE
Allow setting debug mode via CLI

### ISSUE THREAD

**Issue by hackaugusto** [0xPolygonMiden/miden-vm#1322]
### Task

Add a flag to the run command which enables the assembler debug mode.

**Comment by bobbinth** [comment]
We already debug mode in CLI (see [here](https://0xpolygonmiden.github.io/miden-vm/tools/debugger.html)). Or did you mean something else?

**Comment by yasonk** [comment]
Would this `--debug` argument be passed into the command like this after building?
```
./target/optimized/miden run -a miden/examples/nprime/nprime.masm --debug
```

Why not just do this?
```
./target/optimized/miden debug -a miden/examples/nprime/nprime.masm
```

This seems to already support `debug.stack`

**Comment by bobbinth** [comment]
> Would this `--debug` argument be passed into the command like this after building?
> 
> ```
> ./target/optimized/miden run -a miden/examples/nprime/nprime.masm --debug
> ```

Yes, that's how I was thinking it would work.

> Why not just do this?
> 
> ```
> ./target/optimized/miden debug -a miden/examples/nprime/nprime.masm
> ```

This command would start an interactive debugger and then the user would be able to step through the execution. The two approaches are similar but the interactive debugger is quite a bit heavier and may not be needed in all cases.

**Comment by bobbinth** [comment]
Closed by #1502.
